adapter/internal/oasparser/model/open_api.go

```diff
@@ -346,6 +346,12 @@ func getEndpointForWebsocketURL(rawURL string) (*Endpoint, error) {
 		return nil, err
 	}
 
+	// Hostname validation
+	if err == nil && !regexp.MustCompile(hostNameValidator).MatchString(parsedURL.Hostname()) {
```

Comment:
We don't want to check err == nil here.
In line 344, if err != nill, return form the func.